### ISSUE TITLE
[Enhancement]: A11y improvements for Page (set html title + focus h1 title on page enter)

### ISSUE THREAD

**Issue by jakobe** [kirbydesign/designsystem#3692]
`Page` component should:

- [ ] Set the html document's `title` property (using Angular's `Title` service) to the title of the page (`Page.Title` prop, `Header.Title` prop or from a custom `PageTitleDirective`
   - [ ] Should also support async titles (e.g. when using `*ngIf` or `| async` pipe
- [ ] Focus the `H1` element on page enter
   - [ ] `H1` element should have `tabindex="-1"` to allow setting `focus()`
